### PR TITLE
Auth changes & remove api changes

### PR DESCRIPTION
**Pull request by sivakumar-kailasam**
- Updated docs on local & firebase settings
- Make remove api from data store update the existing data instead of removing it for future analysis
- Authenticate with firebase and reauthenticate on token expiry

Implements #8

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,12 +3,35 @@ Used to populate our backend (currently Firebase) for the Ember Help Wanted app
...
   - Payload URL: __YourNGrokUrl__/issue-handler
   - Secret: oursecrethere
 - To inspect payload visit http://127.0.0.1:4000
 - To inspect logs run ./node_modules/.bin/bunyan logs/events.log
 
+### Environment variables to setup
+```
+export OPENSHIFT_NODEJS_IP="127.0.0.1"
+export OPENSHIFT_NODEJS_PORT=8080
+export FIREBASE_APP="https://help-wanted-dev.firebaseio.com"
```
Comment (acorncom) [inline README.md]: These shouldn't need to be set unless they are different

File: README.md [hunk 2]
```diff
@@ -3,12 +3,35 @@ Used to populate our backend (currently Firebase) for the Ember Help Wanted app
...
+export FIREBASE_SECRET="344SliTRyGVneG8QdRfCKRjF6wlkzFoHlpjNYHqY"
```
Comment (acorncom) [inline README.md]: Probably better to not have this exposed? You might want to reset the key too ...

File: README.md (hunk 1 above)
Comment (sivakumar-kailasam) [inline README.md]: so these are sample values, we can remove the whole section. but thought i'd save people time 😄

File: package.json [hunk 3]
```diff
@@ -24,17 +24,18 @@
...
   "homepage": "https://github.com/ember-learn/ember-help-wanted-webhook",
   "devDependencies": {
-    "mocha": "^2.4.5",
-    "mocha-only-detector": "^0.1.0",
-    "sinon": "^1.17.3"
+    "bluebird": "3.3.5",
+    "bunyan": "1.8.1",
+    "firebase": "^2.4.2",
+    "firebase-token-generator": "2.0.0",
+    "github-webhook-handler": "^0.5.0"
```
Comment (acorncom) [inline package.json]: These have to be dependencies not dev dependencies in order for openstack to launch (see my earlier commit)

File: README.md (hunk 2 above)
Comment (sivakumar-kailasam) [inline README.md]: the one actually set in dev is different

File: README.md (hunk 1 above)
Comment (acorncom) [inline README.md]: Could we just show the items they'll need to setup? I like the idea of saving time :smile: but seems like a ton of things to setup when they actually might only need Firebase info ;-)

File: README.md (hunk 2 above)
Comment (acorncom) [inline README.md]: Ah, got it. Maybe your secret here or something so that's clearer?

**Comment by acorncom** [comment]
Thanks!

**Pull request merged**